### PR TITLE
Add general WHATWG guidelines for contributors and maintainers

### PR DESCRIPTION
**Pull request by foolip**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+These are general guidelines for contributing for all WHATWG standards.
+
+### Pull requests
+
+Leave the **Allow edits from maintainers** option enabled to allow reviewers to fix trivial issues directly on your branch rather than needing to write review comments asking you make the edits.  For more details, see [Improving collaboration with forks](https://github.com/blog/2247-improving-collaboration-with-forks) in the GitHub Blog.
+
+### Tests
+
+For normative changes, a corresponding [web-platform-tests](https://github.com/w3c/web-platform-tests) PR is highly appreciated. The author and reviewer can be different from the spec PR. If current behavior is unclear, writing tests first can help inform the discussion. Typically, both PRs will be merged at the same time.
```
Comment (foolip) [inline CONTRIBUTING.md]: @annevk, here I had to change the wording from "the HTML Standard PR" to "the spec PR" and I remember you didn't want this wording originally. Thoughts?

**Comment by foolip** [comment]
Let's use this PR for the general discussion. This splits our guidelines into multiple documents, which isn't great for discoverability, but I don't see how else to do it.

File: CONTRIBUTING.md (hunk 1 above)
Comment (annevk) [inline CONTRIBUTING.md]: "the PR for the standard"?

Comment (foolip) [inline CONTRIBUTING.md]: Thanks, now updated.

**Comment by zcorpan** [comment]
Should we move (part of) TEAM.md here as well?

**Comment by foolip** [comment]
OK, I've moved almost all of TEAM.md as well. I ended up doing some tinkering for consistency of headings and alike, PTAL.

File: TEAM.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+Each change needs to result in a single commit on the master branch, with no merge commits. The green squash and merge button is OK to use, but be sure to tidy up the commit message per [guidelines for writing good commit messages](https://github.com/erlang/otp/wiki/Writing-good-commit-messages).
+
+Prefix the summary line with "Editorial: " if the change is just to fix formatting, typos, or is a refactoring that does not change how the standard is understood. Note that bug fixes or clarifications are not editorial, even if they only affect non-normative text.
+
+Prefix the summary line with "Meta: " for changes that do not directly affect the text of the standard, but instead the ecosystem around it, such as spec tooling or contributor documentation.
+
+For normative changes, ask for a [web-platform-tests](https://github.com/w3c/web-platform-tests) PR if testing is practical and not overly burdensome. Aim to merge both PRs at the same time. If one PR is approved but the other needs more work, add the `do not merge yet` label.
+
+If a follow-up issue is filed, add the appropriate label, e.g. `html` for the HTML Standard.
```
Comment (domenic) [inline TEAM.md]: "filed in the web-platform-tests repository" for clarity

File: TEAM.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+### Checking out pull requests from forks
+
+Pull requests from external contributors come from branches in their forks. You can check out those external branches in order to review and test the commits in those pull requests, and to be able to push changes to them on your own (e.g., fixes for typos)—rather than needing to write review comments asking the PR contributor to make the edits.
+
+To checkout a PR branch, note the user it's coming from and the branch they used in their fork. For example, for user `estark37` with branch `example-fix`, you would do
```
Comment (domenic) [inline TEAM.md]: "on the `html` repository"

**Comment by foolip** [comment]
Squashed into a single commit for easy merging once https://github.com/whatwg/html/pull/2295 is OK.

File: TEAM.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+checkout-pr estark37 example-fix
+```
+
+### Merging pull requests into master
+
+Just use the normal green button in the pull-request page in the GitHub Web UI, but first ensure the commit message follows [the guidelines](https://github.com/erlang/otp/wiki/Writing-good-commit-messages).
+
+#### Merging to master from the command line
+
+Regardless of whether a pull request comes from a contributor (in which case the branch is from a different remote repository and you've already used the **Checking out pull requests from forks** steps above to fetch the branch) or from other editors or members of the WHATWG GitHub organization (in which case the branch is within this repository), the steps for cleanly merging it to master are the same:
```
Comment (zcorpan) [inline TEAM.md]: "this repository" is now not correct

Comment (foolip) [inline TEAM.md]: I'll call it "the main repository"

**Pull request merged**